lib/util/linkComponents.js

```diff
@@ -3,6 +3,7 @@
  */
 'use strict';
 
+/** @type {any[]}} */
```

Comment:
`string[]`?

Reply:
Tsc does not accept `string[]`. The official type declaration requires that in `a.concat(b)`, `a` and `b` must have the same type.

Reply:
when is this array concatted with something that's not also a `string[]`?

Reply:
In lines bellow it is concatted with `setting.something`.

Reply:
which should be a `string[]` per the schema, no?

Reply:
By schema it could be `{name: something, linkAttribute: something}[]`

Reply:
ah - then should the type be `(string | { name: string, linkAttribute: string })[]`?

Reply:
But `settings.linkComponents` is not validated, it could be anything.

Reply:
if it's not that schema tho, the plugin won't work properly, so i think we can assume that's its type.